### PR TITLE
:seedling: refine placement helper function

### PR DESCRIPTION
**Pull request by haoqing0110**
## Related issue(s)

Fixes #

### CODE REVIEW

File: cluster/v1beta1/helpers.go [hunk 1]
```diff
@@ -217,83 +234,41 @@ func (pdct *PlacementDecisionClustersTracker) generateGroupsNameIndex() {
...
-		existingClusters.Insert(clusterSets.UnsortedList()...)
-	}
-
-	return existingClusters, existingGroupKeys, existingClusterGroups
-}
-
-// ExistingClusterGroups returns the tracker's existing decision cluster names for groups listed in groupKeys.
-// Return empty set when groupKeys is empty.
-func (pdct *PlacementDecisionClustersTracker) ExistingClusterGroups(groupKeys []GroupKey) ([]GroupKey, map[GroupKey]sets.Set[string]) {
+func (pdct *PlacementDecisionClustersTracker) ExistingClusterGroups(groupKeys []GroupKey) ClusterGroupsMap {
```
Comment (qiujian16) [inline cluster/v1beta1/helpers.go]: can we use ExistingClusterGroups(groupKeys ...GroupKey)?

File: cluster/v1beta1/helpers.go [hunk 2]
```diff
@@ -317,16 +292,56 @@ func (pdct *PlacementDecisionClustersTracker) fulfillGroupKeys(groupKeys []Group
...
+	// sort by group index index
+	sort.Slice(groupKeys, func(i, j int) bool {
+		return groupKeys[i].GroupIndex < groupKeys[j].GroupIndex
+	})
+
+	return groupKeys
+}
+
+// GetClusterSets returns a set containing all clusters from all group sets.
+func (g ClusterGroupsMap) GetClusterSets() sets.Set[string] {
```
Comment (qiujian16) [inline cluster/v1beta1/helpers.go]: ClusterSets here is confusing, GetClusters maybe

File: cluster/v1beta1/helpers.go (hunk 1 above)
Comment (haoqing0110) [reply cluster/v1beta1/helpers.go]: Have updated the code.

File: cluster/v1beta1/helpers.go (hunk 2 above)
Comment (haoqing0110) [reply cluster/v1beta1/helpers.go]: Fixed.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**